### PR TITLE
36 search card sequence editor

### LINKED ISSUE #36: ECS.8 Search for card in sequence editor
## User Story 
As a user I want a search bar, to easily find and select the card I want in a sequence

## Sub-tasks 
- [x] Search bar in the card lists
- [x] Filter function
- [x] As the user type in the search bar, cards are updated to match the search

## Notes
- depends on #31 
- Look into using array.includes(substring) to filter cards
- I would do this before you do #35

### PR DESCRIPTION
**Pull request by euilee0129**
fixes #36
Search bar
- utilized pipe called filter, filters search based on words searched in bar
- border get darkened, changes color to signify search bar is clicked and being used

### CODE REVIEW

File: src/app/sequence-editor/_sequence-editor-theme.scss [hunk 1]
```diff
@@ -85,4 +85,8 @@
   .selected-list-item {
     background-color: mat.get-theme-color($theme, secondary-container);
   }
+  #search-bar:focus {
+    outline: 1px solid #ddf0c8;
+    background-color: #dcefc7;
+  }
```
Comment (ashkjones) [inline src/app/sequence-editor/_sequence-editor-theme.scss:91]: Could you combine the #search-bar and .card-search-bar styles like I asked on discord?

File: src/app/sequence-editor/_sequence-editor-theme.scss [hunk 2]
```diff
@@ -85,4 +85,8 @@
   .selected-list-item {
     background-color: mat.get-theme-color($theme, secondary-container);
   }
+  #search-bar:focus {
+    outline: 1px solid #ddf0c8;
+    background-color: #dcefc7;
```
Comment (ashkjones) [inline src/app/sequence-editor/_sequence-editor-theme.scss:90]: I don't think the color is distinctive enough. Also, use a color from the theme, not a hard coded color.

image

**Review by ashkjones — CHANGES_REQUESTED** [review]
Good start. I am requesting some code style changes and that you use colors from the theme

File: src/app/sequence-editor/sequence-editor.component.html [hunk 3]
```diff
@@ -2,9 +2,9 @@
   <div class="sequence-tab-left">
     <div class="sequence-card-list">
       <h3 class="sequence-section-title">All Cards</h3>
-      <div class="card-search-bar">Search for a card here...</div>
+        <input class="card-search-bar" #txt (input)="filterItems(txt.value)"type="search" [(ngModel)]="searchtext" id = "search-bar"placeholder="search" aria-label="search">
       <ol class="card-list">
-        @for( flashcard of _flashcards; track _flashcards){
+        @for( flashcard of filteredCards; track filteredCards){
```
Comment (ashkjones) [inline src/app/sequence-editor/sequence-editor.component.html:7]: Try to keep line lengths under 100 characters. You can check in the bottom right of vs code
image

File: src/app/sequence-editor/sequence-editor.component.ts [hunk 4]
```diff
@@ -117,4 +126,8 @@ export class SequenceEditorComponent {
       moveItemInArray(this.selectedSequence.cardList, event.previousIndex, event.currentIndex);
     }
   }
+  
+  filterItems(filterValue: string) {
+    this.filteredCards = this.flashcards.filter(item => item.term.toLowerCase().includes(filterValue.toLowerCase()));
```
Comment (ashkjones) [inline src/app/sequence-editor/sequence-editor.component.ts:131]: Try to keep line lengths under 100 characters

File: src/app/filter.pipe.spec.ts
Comment (ashkjones) [inline src/app/filter.pipe.spec.ts:1]: I don't think you ever use this

File: src/app/filter.pipe.ts
Comment (ashkjones) [inline src/app/filter.pipe.ts:1]: i don't think you use this file either

File: src/app/sequence-editor/sequence-editor.component.ts [hunk 5]
```diff
@@ -16,25 +16,27 @@ import {
...
+        MatButtonToggleModule,
+        MatCardModule,
+        MatIconModule,
+        FlashcardComponent,
+        EcCardPreviewComponent,
+        DragDropModule,
+        MatInputModule,
+        MatFormFieldModule,
+        FormsModule,
+        FilterPipe
```
Comment (ashkjones) [inline src/app/sequence-editor/sequence-editor.component.ts:38]: remove the filter pip because it isn't used

**Review by ashkjones — CHANGES_REQUESTED** [review]

**Review by iamdylanhuynh001 — APPROVED** [review]
Search bar searches correctly.

Seems good to me

**Review by ashkjones — APPROVED** [review]

**Pull request merged**